File: ExpressionTreeTransform.Tests.VB/CompilerGenerated/Literals.vb
Public Class Literals
    <Fact>
    Sub [True]()
        BuildAssert(Function() True, "() => true", "Function() True")
    End Sub

    <Fact>
    Sub [False]()
        BuildAssert(Function() False, "() => false", "Function() False")
    End Sub

    <Fact>
    Sub [Nothing]()
        BuildAssert(Function() CType(Nothing, String), "() => null", "Function() Nothing")
    End Sub

    <Fact>
    Sub [Integer]()
        BuildAssert(Function() 5, "() => 5", "Function() 5")
    End Sub

    <Fact>
    Sub NonInteger()
        BuildAssert(Function() 7.32, "() => 7.32", "Function() 7.32")
    End Sub

    <Fact>
    Sub [String]()
        BuildAssert(Function() "abcd", "() => ""abcd""", "Function() ""abcd""")
    End Sub

    <Fact>
    Sub DateTime()
        BuildAssert(Function() #1981-1-1#, "() => #DateTime", $"Function() #{#1981-1-1#.ToString()}#")
    End Sub

    <Fact>
    Sub TimeSpan()
        Dim ts = New TimeSpan(5, 4, 3, 2, 1)
        BuildAssert(Constant(ts), "#TimeSpan", $"#{ts.ToString()}#")
    End Sub
End Class

File: ExpressionToString.Tests/CompilerGenerated/Literals.cs
using System;
using static System.Linq.Expressions.Expression;
using Xunit;
using static ExpressionToString.Tests.Runners;
using static ExpressionToString.Tests.Globals;

namespace ExpressionToString.Tests {
    [Trait("Source", CSharpCompiler)]
    public class Literals {
        [Fact]
        public void True() => RunTest(() => true, "() => true", "Function() True");

        [Fact]
        public void False() => RunTest(() => false, "() => false", "Function() False");

        [Fact]
        public void Nothing() => RunTest(() => (string)null, "() => null", "Function() Nothing");

        [Fact]
        public void Integer() => RunTest(() => 5, "() => 5", "Function() 5");

        [Fact]
        public void NonInteger() => RunTest(() => 7.32, "() => 7.32", "Function() 7.32");

        [Fact]
        public void String() => RunTest(() => "abcd", "() => \"abcd\"", "Function() \"abcd\"");

        [Fact]
        public void EscapedString() => RunTest(
            () => "\'\"\\\0\a\b\f\n\r\t\v",
            @"() => ""\'\""\\\0\a\b\f\n\r\t\v""",
            "Function() \"\'\"\"\\\0\a\b\f\n\r\t\v\""
        );

        [Fact]
        public void InterpolatedString() => RunTest(
            () => $"{new DateTime(2001, 1, 1)}",
            "() => $\"{(object)new DateTime(2001, 1, 1)}\"",
            "Function() $\"{CObj(New Date(2001, 1, 1))}\""
        );

        [Fact]
        public void Type() => RunTest(
            () => typeof(string),
            "() => typeof(string)",
            "Function() GetType(String)"
        );
    }
}
